src/Optical/LensAssembly.jl

```diff
@@ -50,37 +53,16 @@ macro lensassembly_constructor(N)
     return esc(quote
         struct $(name){T,$(typesig...)} <: LensAssembly{T}
             axis::SVector{3,T}
-            rectangles::Vector{Rectangle{T}}
-            ellipses::Vector{Ellipse{T}}
-            hexagons::Vector{Hexagon{T}}
+            planarshapes::Vector{PlanarShape{T}}
             paraxials::Vector{ParaxialLens{T}}
             $(fields...)
         end
     end)
 end
 
-# This is the basic idea of what each closestintersection implementation does:
```

Comment:
I left this commented code originally because it is quite hard to parse the logic just from reading through the macros below - for someone new to the codebase I think it is useful to keep it

Reply:
Fair enough, I can add it back in. Although I've actually further cleaned up the code in this other branch https://github.com/microsoft/OpticSim.jl/commit/ad066db43a0b0e70e8e4965c96ffc9203d358c0b. It's not quite the same though - do you think this is explanatory enough?